oss-fuzz: artifacts.(*ArtifactServer).GetInputArtifactByUID panic

https://oss-fuzz.com/testcase-detail/5098631152140288

```
panic: runtime error: index out of range [2] with length 2
goroutine 17 [running, locked to thread]:
github.com/argoproj/argo-workflows/v3/server/artifacts.(*ArtifactServer).getArtifactByUID(0x10c0004a4780, {0x2b3f5c0, 0x10c0002d56a0}, 0x10c000559d00, 0x1)
 github.com/argoproj/argo-workflows/v3/server/artifacts/artifact_server.go:101 +0x5d2
github.com/argoproj/argo-workflows/v3/server/artifacts.(*ArtifactServer).GetInputArtifactByUID(...)
 github.com/argoproj/argo-workflows/v3/server/artifacts/artifact_server.go:95
github.com/argoproj/argo-workflows/v3/server/artifacts.FuzzGetOutputArtifactByUID({0x602000000090, 0x0, 0x1})
 github.com/argoproj/argo-workflows/v3/server/artifacts/artifacts_fuzzer.go:61 +0x35c
```

--- a/server/artifacts/artifact_server.go
+++ b/server/artifacts/artifact_server.go
@@ -96,8 +96,11 @@ func (a *ArtifactServer) GetInputArtifactByUID(w http.ResponseWriter, r *http.Re
 }
 
 func (a *ArtifactServer) getArtifactByUID(w http.ResponseWriter, r *http.Request, isInput bool) {
-	requestPath := strings.SplitN(r.URL.Path, "/", 6)
-
+	requestPath := strings.SplitN(r.URL.Path, "/", 5)
+	if len(requestPath) != 5 {
+		a.serverInternalError(errors.New("request path is not valid"), w)
+		return
+	}
 	uid := requestPath[2]
 	nodeId := requestPath[3]
 	artifactName := requestPath[4]

--- a/server/artifacts/artifact_server_test.go
+++ b/server/artifacts/artifact_server_test.go
@@ -231,3 +231,20 @@ func TestArtifactServer_GetOutputArtifactByUID(t *testing.T) {
 	s.GetOutputArtifactByUID(w, r)
 	assert.Equal(t, 401, w.StatusCode)
 }
+
+func TestArtifactServer_GetArtifactByUIDInvalidRequestPath(t *testing.T) {
+	s := newServer()
+	r := &http.Request{}
+	// missing my-artifact part to have a valid URL
+	r.URL = mustParse("/input-artifacts/my-uuid/my-node")
+	w := &testhttp.TestResponseWriter{}
+	s.GetInputArtifactByUID(w, r)
+	// make sure there is no index out of bounds error
+	assert.Equal(t, 500, w.StatusCode)
+	assert.Equal(t, "request path is not valid", w.Output)
+
+	w = &testhttp.TestResponseWriter{}
+	s.GetOutputArtifactByUID(w, r)
+	assert.Equal(t, 500, w.StatusCode)
+	assert.Equal(t, "request path is not valid", w.Output)
+}

